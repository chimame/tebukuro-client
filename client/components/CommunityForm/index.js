import React, { Component } from 'react'

const onErrorComponent = (community) => {
  if (community.isError()) {
    let errors = []

    community.errors.forEach((error) => {
      errors.push(<p key={ error }>{error}</p>)
    })

    return <div className="error">{ errors }</div>
  }
}

export default class CommunityForm extends Component {
  onSubmitHandler(e) {
    e.preventDefault()
    const name = this.refs.inputName.value
    const description = this.refs.inputDescription.value
    this.props.onSubmit(name, description)
  }

  render() {
    return (
      <div>
        <h3>create community form</h3>
        { onErrorComponent(this.props.community) }
        <form onSubmit={ this.onSubmitHandler.bind(this) }>
          <div>
            <label>
              name
              <input type="text" ref="inputName" />
            </label>
          </div>
          <div>
            <label>
              description
              <textarea ref="inputDescription" />
            </label>
          </div>
          <div>
            <input type="submit" />
          </div>
        </form>
      </div>
    )
  }
}

CommunityForm.propTypes = {
  onSubmit: React.PropTypes.func.isRequired,
  community: React.PropTypes.object.isRequired
}
