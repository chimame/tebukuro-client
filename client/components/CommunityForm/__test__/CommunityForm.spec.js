import React from 'react';
import CommunityForm from '../index'
import renderer from 'react-test-renderer'

describe('CommunityForm', () => {
  it('render form', () => {
    const component = renderer.create(
      <CommunityForm />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
