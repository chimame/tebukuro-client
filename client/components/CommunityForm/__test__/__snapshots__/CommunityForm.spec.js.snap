exports[`CommunityForm render form 1`] = `
<div>
  <h3>
    create community form
  </h3>
  <form>
    <div>
      <label>
        name
        <input
          type="text" />
      </label>
    </div>
    <div>
      <label>
        description
        <textarea />
      </label>
    </div>
    <div>
      <input
        type="submit" />
    </div>
  </form>
</div>
`;
